internal/packages/packages.go

```diff
@@ -5,94 +5,63 @@
 package packages
 
 import (
-	"encoding/json"
-	"io/ioutil"
 	"os"
 	"path/filepath"
 
+	"github.com/elastic/go-ucfg"
+	"github.com/elastic/go-ucfg/yaml"
 	"github.com/pkg/errors"
-	"gopkg.in/yaml.v3"
 )
 
 const (
 	// PackageManifestFile is the name of the package's main manifest file.
 	PackageManifestFile = "manifest.yml"
 
-	// DataStreamManifestFile is the name of the dataStream's manifest file.
+	// DataStreamManifestFile is the name of the data stream's manifest file.
 	DataStreamManifestFile = "manifest.yml"
 )
 
-// VarValue represents a variable value as defined in a package or dataStream
-// manifest file.
-type VarValue struct {
-	scalar string
-	list   []string
+// Variable is an instance of configuration variable (named, typed).
+type Variable struct {
+	Name    string      `config:"name" json:"name" yaml:"name"`
+	Type    string      `config:"type" json:"type" yaml:"type"`
+	Default interface{} `config:"default" json:"default" yaml:"default"`
 }
 
-// UnmarshalYAML knows how to parse a variable value from a package or dataStream
-// manifest file into a VarValue.
-func (vv *VarValue) UnmarshalYAML(value *yaml.Node) error {
-	switch value.Kind {
-	case yaml.ScalarNode:
-		vv.scalar = value.Value
-	case yaml.SequenceNode:
-		vv.list = make([]string, len(value.Content))
-		for idx, content := range value.Content {
-			vv.list[idx] = content.Value
-		}
-	default:
-		return errors.New("unknown variable value")
-	}
-	return nil
-}
-
-// MarshalJSON knows how to serialize a VarValue into the appropriate
-// JSON data type and value.
-func (vv VarValue) MarshalJSON() ([]byte, error) {
-	if vv.scalar != "" {
-		return json.Marshal(vv.scalar)
-	} else if vv.list != nil {
-		return json.Marshal(vv.list)
-	}
-	return nil, nil
-}
-
-type variable struct {
-	Name    string   `json:"name"`
-	Type    string   `json:"type"`
-	Default VarValue `json:"default"`
-}
-
-type input struct {
-	Type string     `json:"type"`
-	Vars []variable `json:"vars"`
+// Input is a single input configuration.
+type Input struct {
+	Type string     `config:"type" json:"type" yaml:"type"`
+	Vars []Variable `config:"vars" json:"vars" yaml:"vars"`
 }
 
-type policyTemplate struct {
-	Inputs []input `json:"inputs"`
+// PolicyTemplate is a configuration of inputs responsible for collecting log or metric data.
+type PolicyTemplate struct {
```

Comment:
Is the visibility change needed for `ucfg` to work correctly?

Reply:
It's due to consistency. If somebody wants to use Manifest structure and extract policy templates to a separate variable, it wouldn't be possible (can't use a package visible field). In general exposed fields should use public structures.